diff -ru rocm/rocprim/include/rocprim/types/integer_sequence.hpp rocm/rocprim/include/rocprim/types/integer_sequence.hpp
--- rocm/rocprim/include/rocprim/types/integer_sequence.hpp	2019-08-27 17:47:02.426632521 +0000
+++ rocm/rocprim/include/rocprim/types/integer_sequence.hpp	2019-08-27 17:47:06.474637231 +0000
@@ -29,6 +29,7 @@
 #if defined(__cpp_lib_integer_sequence) && !defined(DOXYGEN_SHOULD_SKIP_THIS)
 // For C++14 or newer we just use standard implementation
 using std::integer_sequence;
+using std::index_sequence;
 using std::make_integer_sequence;
 using std::make_index_sequence;
 using std::index_sequence_for;
