diff -ru boost/mpi/collectives/all_reduce.hpp boost/mpi/collectives/all_reduce.hpp
--- boost/mpi/collectives/all_reduce.hpp	2017-12-20 14:26:58.105542000 +0100
+++ boost/mpi/collectives/all_reduce.hpp	2017-12-20 14:56:39.862227286 +0100
@@ -51,7 +51,8 @@
                   T* out_values, Op op, mpl::false_ /*is_mpi_op*/,
                   mpl::true_ /*is_mpi_datatype*/)
   {
-    user_op<Op, T> mpi_op(op);
+    static auto lop = op;
+    user_op<Op, T> mpi_op(lop);
     BOOST_MPI_CHECK_RESULT(MPI_Allreduce,
                            (const_cast<T*>(in_values), out_values, n,
                             boost::mpi::get_mpi_datatype<T>(*in_values),
diff -ru boost/mpi/collectives/scatter.hpp boost/mpi/collectives/scatter.hpp
--- boost/mpi/collectives/scatter.hpp	2017-12-20 14:26:58.079548000 +0100
+++ boost/mpi/collectives/scatter.hpp	2017-12-20 15:07:44.242734024 +0100
@@ -69,7 +69,7 @@
         // Send archive
         packed_oarchive oa(comm);
         for (int i = 0; i < n; ++i)
-          oa << in_values[dest * n + i];
+          oa << in_values[dest * n + i]; // NOLINT
         detail::packed_archive_send(comm, dest, tag, oa);
       }
     }
diff -ru boost/variant/detail/forced_return.hpp boost/variant/detail/forced_return.hpp
--- boost/variant/detail/forced_return.hpp	2017-12-20 14:26:13.518182000 +0100
+++ boost/variant/detail/forced_return.hpp	2017-12-20 16:24:49.834173291 +0100
@@ -39,7 +39,7 @@
     BOOST_ASSERT(false);
 
     T (*dummy_function_ptr)() = 0;
-    return dummy_function_ptr();
+    return dummy_function_ptr(); // NOLINT
 }
 
 template <>
