--- thrust/system/cuda/detail/bulk/detail/synchronize.hpp	2017-10-19 10:57:38.732756117 +0200
+++ thrust/system/cuda/detail/bulk/detail/synchronize.hpp	2018-01-18 17:26:10.922050554 +0100
@@ -29,7 +29,7 @@
 {
 
 
-inline __host__ __device__
+inline __host__
 void synchronize(const char* message = "")
 {
 #if __BULK_HAS_CUDART__
@@ -41,7 +41,7 @@
 } // end terminate()
 
 
-inline __host__ __device__
+inline __host__
 void synchronize_if_enabled(const char* message = "")
 {
 // XXX we rely on __THRUST_SYNCHRONOUS here
--- thrust/system/cuda/detail/synchronize.h	2017-10-19 10:57:38.664757761 +0200
+++ thrust/system/cuda/detail/synchronize.h	2018-01-18 17:25:17.049664809 +0100
@@ -29,14 +29,14 @@
 {
 
 
-inline __host__ __device__
+inline __host__
 void synchronize(const char *message = "");
 
-inline __host__ __device__
+inline __host__
 void synchronize(cudaStream_t stream, const char *message = "");
 
 
-inline __host__ __device__
+inline __host__
 void synchronize_if_enabled(const char *message = "");
 
 
--- thrust/system/cuda/detail/synchronize.inl	2017-10-19 10:57:38.676757471 +0200
+++ thrust/system/cuda/detail/synchronize.inl	2018-01-18 17:25:06.761591137 +0100
@@ -29,14 +29,14 @@
 {
 
 
-inline __host__ __device__
+inline __host__
 void synchronize(const char *message)
 {
   throw_on_error(cudaDeviceSynchronize(), message);
 } // end synchronize()
 
 
-inline __host__ __device__
+inline __host__
 void synchronize(cudaStream_t stream, const char *message)
 {
 #if !defined(__CUDA_ARCH__)
@@ -46,7 +46,7 @@
 #endif
 }
 
-inline __host__ __device__
+inline __host__
 void synchronize_if_enabled(const char *message)
 {
 // XXX this could potentially be a runtime decision
